File: Exemplos SmartPOS/App_IntentDigitalHUB_Java/Elgin_IntentDigitalHUB_SmartPOS_Java-font/app/src/main/java/com/elgin/java_intentdigitalhub_smartpos/IntentDigitalHubService/ELGINPAY/Commands/IniciaCancelamentoVenda.java
package com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.ELGINPAY.Commands;

import androidx.annotation.NonNull;

import com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.ELGINPAY.ElginPayCommand;
import com.google.gson.JsonObject;

public class IniciaCancelamentoVenda extends ElginPayCommand {
    final private String valorTotal;
    final private String ref;
    final private String data;

    public IniciaCancelamentoVenda(@NonNull String valorTotal, @NonNull String ref, @NonNull String data) {
        super("iniciaCancelamentoVenda");
        this.valorTotal = valorTotal;
        this.ref = ref;
        this.data = data;
    }

    @Override
    protected JsonObject functionParametersJson() {
        JsonObject functionParametersJson = new JsonObject();

        functionParametersJson.addProperty("valorTotal", this.valorTotal);
        functionParametersJson.addProperty("ref", this.ref);
        functionParametersJson.addProperty("data", this.data);

        return functionParametersJson;
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/IntentDigitalHubService/ELGINPAY/Commands/IniciaCancelamentoVenda.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.ELGINPAY.Commands

import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.ELGINPAY.ElginPayCommand
import com.google.gson.JsonObject

class IniciaCancelamentoVenda(
    private val valorTotal: String,
    private val ref: String,
    private val data: String
) :
    ElginPayCommand("iniciaCancelamentoVenda") {
    override fun functionParametersJson(): JsonObject {
        val functionParametersJson = JsonObject()
        functionParametersJson.addProperty("valorTotal", valorTotal)
        functionParametersJson.addProperty("ref", ref)
        functionParametersJson.addProperty("data", data)
        return functionParametersJson
    }
}

